code.sh

```diff
@@ -61,4 +61,4 @@ for i in "${SDK[@]}"; do
 done
 
 exec env PATH="${PATH}:${XDG_DATA_HOME}/node_modules/bin" \
-  /app/extra/vscode/bin/code --extensions-dir=${XDG_DATA_HOME}/vscode/extensions "$@" ${WARNING_FILE}
+  /app/bin/zypak-wrapper.sh /app/extra/code "$@" --extensions-dir=${XDG_DATA_HOME}/vscode/extensions "$@" ${WARNING_FILE}
```

Comment:
By running the electron binary directly, you completely omit the .js entrypoint responsible for handling cli args.

Reply:
Can you explain what this means?

Anyhow, Zypak is supposed to directly run the binary, and not the wrapper.

Reply:
> Can you explain what this means?

This means that cli args won't work. Check with `flatpak run com.visualstudio.code --help`. I can't say if anything else will break.

> Zypak is supposed to directly run the binary, and not the wrapper.

Yes, so you'll need to reimplement relevant parts of the upstream wrapper and put in into flatpak's one.

Reply:
I think I got it working.

Reply:
Hmm... I thought it worked, but it didn't.

Reply:
@gasinvein do you have an idea how we could reimplement? I can't get it to work.

Reply:
You've probably forgot `ELECTRON_RUN_AS_NODE=1`.

Reply:
Also, the really tricky part is passing the cli handler only to the initial process, but not to its reexeced children (zygotes or whatever they're called).
I didn't find any other way to use two wrappers: 1st, "outer", passes the cli handler arg, `--extensions-dir` and `ELECTRON_RUN_AS_NODE`, while the 2nd, "inner", is the normal one that sets up development environment, without passing any additional args; it's the 2nd which is used as `CHROME_WRAPPER`.

I've did mostly the same for Code-OSS, and it mostly works, but still decided to stay with `--no-sandbox` for now. I can push the relevant changes to the git repo if you'd like.

Reply:
> You've probably forgot ELECTRON_RUN_AS_NODE=1.

I forgot to mention that I tried it locally but it didn't work. When I tried without it, I improperly tested it which made me think it worked, but it didn't either, so I just reverted the commit.

> I've did mostly the same for Code-OSS, and it mostly works, but still decided to stay with --no-sandbox for now. I can push the relevant changes to the git repo if you'd like.

What do you mean by relevant changes? Are you talking about my patches?

Reply:
I mean, I have changes for integrating zypak into code-oss locally, but didn't push them to the flathub git repo.

Reply:
Ah okay. If you can push them in a new branch, that would be nice.

Reply:
[Here](https://github.com/flathub/com.visualstudio.code-oss/commit/42d4f1baa3d9e17a4cd866556ecb2471b5bdceab) it is.

Reply:
> I've did mostly the same for Code-OSS, and it mostly works, but still decided to stay with --no-sandbox for now. I can push the relevant changes to the git repo if you'd like.

When you say "it mostly works", does it mean that there are some things that do not work or does it mean something else?

Reply:
From what I noticed, `--wait` option doesn't work. Didn't yet figure out why.